### ISSUE TITLE
Randomize MAC for Apple BLE spam?

### ISSUE THREAD

**Issue by MatthewGuidry** [ClaraCrazy/Flipper-Xtreme#380]
### Describe the enhancement you're suggesting.

In my limited testing you only get one popup for each device type before some kind of cooldown period expires on the Apple device. Perhaps having a different hardware address would allow this to be circumvented? Even if that's not the case, it would be nice to have these spam messages to originate from an address that differs from the Flipper's default so that they're not as easily associated. 

One other minor thing would be to turn off bluetooth when exiting the app if it was wasn't enabled when it was opened.

I know the BLE spam is super new and only in the dev branch so far, so maybe these are things you're already thinking about.

**Comment by Willy-JL** [comment]
> so maybe these are things you're already thinking about

actually already working on all 3 of those things as we speak

**Comment by Willy-JL** [comment]
> Perhaps having a different hardware address would allow this to be circumvented?

proximity pair packets will get another popup if the contained data differs, am yet to dive into proximity pair as ive been investigating more the neraby action packets, which indeed will pop up once then need a lock/unlock to pop up again. one exception is the join this appletv one, which can be dismissed and brought back with the right sequence of packets. i have tried to add in mac address randomization with every single packet, it completely stopped working lol. only the autofill one would somewhat benefit from it, but all the other ones stopped working. so thats a no for now

>  it would be nice to have these spam messages to originate from an address that differs from the Flipper's default so that they're not as easily associated.

this however is now done. it will generate a new random mac address when pressing start, before launching the advertisement beacon with it in background. changing the delay will restart the beacon but keep the mac. pressing stop and start again will make a new mac.

> One other minor thing would be to turn off bluetooth when exiting the app if it was wasn't enabled when it was opened.

fixed now as can be seen in the linked commit, its done at start / stop level, not app. so it will disconnect when starting a flood, and become available again when thats done (but only if it was available before starting)

**Comment by colins770** [comment]
can some one help me get this on xtreem firmware i put the fab file but gives error

**Comment by Willy-JL** [comment]
install the dev update from the discord server

**Comment by herby68** [comment]
Where can I find the BLE Stuff in Xtreme Firmware ? Under Bluetooth ist nothing

**Comment by Constey** [comment]
> > Where can I find the BLE Stuff in Xtreme Firmware ? Under Bluetooth ist nothing
> 
> It's under Apps -> Bluetooth folder -> Apple BLE Spam if your firmware is up to date.

In the latest [XFW-0051_01092023](https://github.com/Flipper-XFW/Xtreme-Firmware/releases/tag/XFW-0051_01092023) it seems not included. I dont see it aswell . Which version you're running?
